Dockerfile cannot be run as is.

The current Dockerfile cannot be run as is without Docker compose.
Note, it works fine with docker-compose.

```sh
$ docker run kilo59/corona-tracker-api
Checking for script in /app/prestart.sh
Running script /app/prestart.sh
Running inside /app/prestart.sh, you could add migrations to this file, e.g.:

#! /usr/bin/env bash

# Let the DB start
sleep 10;
# Run migrations
alembic upgrade head

{"loglevel": "info", "workers": 8, "bind": "0.0.0.0:80", "workers_per_core": 1.0, "host": "0.0.0.0", "port": "80"}
[2020-04-09 02:15:15 +0000] [1] [INFO] Starting gunicorn 20.0.4
[2020-04-09 02:15:15 +0000] [1] [INFO] Listening at: http://0.0.0.0:80 (1)
[2020-04-09 02:15:15 +0000] [1] [INFO] Using worker: uvicorn.workers.UvicornWorker
[2020-04-09 02:15:15 +0000] [8] [INFO] Booting worker with pid: 8
[2020-04-09 02:15:15 +0000] [9] [INFO] Booting worker with pid: 9
[2020-04-09 02:15:15 +0000] [10] [INFO] Booting worker with pid: 10
[2020-04-09 02:15:16 +0000] [11] [INFO] Booting worker with pid: 11
[2020-04-09 02:15:16 +0000] [12] [INFO] Booting worker with pid: 12
[2020-04-09 02:15:16 +0000] [8] [ERROR] Exception in worker process
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/gunicorn/arbiter.py", line 583, in spawn_worker
    worker.init_process()
  File "/usr/local/lib/python3.7/site-packages/uvicorn/workers.py", line 57, in init_process
    super(UvicornWorker, self).init_process()
  File "/usr/local/lib/python3.7/site-packages/gunicorn/workers/base.py", line 119, in init_process
    self.load_wsgi()
  File "/usr/local/lib/python3.7/site-packages/gunicorn/workers/base.py", line 144, in load_wsgi
    self.wsgi = self.app.wsgi()
  File "/usr/local/lib/python3.7/site-packages/gunicorn/app/base.py", line 67, in wsgi
    self.callable = self.load()
  File "/usr/local/lib/python3.7/site-packages/gunicorn/app/wsgiapp.py", line 49, in load
    return self.load_wsgiapp()
  File "/usr/local/lib/python3.7/site-packages/gunicorn/app/wsgiapp.py", line 39, in load_wsgiapp
    return util.import_app(self.app_uri)
  File "/usr/local/lib/python3.7/site-packages/gunicorn/util.py", line 358, in import_app
    mod = importlib.import_module(module)
  File "/usr/local/lib/python3.7/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1006, in _gcd_import
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 967, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 728, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/app/main.py", line 13, in <module>
    from .data import data_source
ImportError: attempted relative import with no known parent package
[2020-04-09 02:15:16 +0000] [8] [INFO] Worker exiting (pid: 8)
[2020-04-09 02:15:16 +0000] [13] [INFO] Booting worker with pid: 13
[2020-04-09 02:15:16 +0000] [14] [INFO] Booting worker with pid: 14
[2020-04-09 02:15:16 +0000] [9] [ERROR] Exception in worker process
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/gunicorn/arbiter.py", line 583, in spawn_worker
    worker.init_process()
  File "/usr/local/lib/python3.7/site-packages/uvicorn/workers.py", line 57, in init_process
    super(UvicornWorker, self).init_process()
  File "/usr/local/lib/python3.7/site-packages/gunicorn/workers/base.py", line 119, in init_process
    self.load_wsgi()
  File "/usr/local/lib/python3.7/site-packages/gunicorn/workers/base.py", line 144, in load_wsgi
    self.wsgi = self.app.wsgi()
  File "/usr/local/lib/python3.7/site-packages/gunicorn/app/base.py", line 67, in wsgi
    self.callable = self.load()
  File "/usr/local/lib/python3.7/site-packages/gunicorn/app/wsgiapp.py", line 49, in load
    return self.load_wsgiapp()
  File "/usr/local/lib/python3.7/site-packages/gunicorn/app/wsgiapp.py", line 39, in load_wsgiapp
    return util.import_app(self.app_uri)
  File "/usr/local/lib/python3.7/site-packages/gunicorn/util.py", line 358, in import_app
    mod = importlib.import_module(module)
  File "/usr/local/lib/python3.7/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1006, in _gcd_import
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 967, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 728, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/app/main.py", line 13, in <module>
    from .data import data_source
ImportError: attempted relative import with no known parent package
[2020-04-09 02:15:16 +0000] [9] [INFO] Worker exiting (pid: 9)
[2020-04-09 02:15:16 +0000] [15] [INFO] Booting worker with pid: 15
[2020-04-09 02:15:16 +0000] [10] [ERROR] Exception in worker process
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/gunicorn/arbiter.py", line 583, in spawn_worker
    worker.init_process()
  File "/usr/local/lib/python3.7/site-packages/uvicorn/workers.py", line 57, in init_process
    super(UvicornWorker, self).init_process()
  File "/usr/local/lib/python3.7/site-packages/gunicorn/workers/base.py", line 119, in init_process
    self.load_wsgi()
  File "/usr/local/lib/python3.7/site-packages/gunicorn/workers/base.py", line 144, in load_wsgi
    self.wsgi = self.app.wsgi()
  File "/usr/local/lib/python3.7/site-packages/gunicorn/app/base.py", line 67, in wsgi
    self.callable = self.load()
  File "/usr/local/lib/python3.7/site-packages/gunicorn/app/wsgiapp.py", line 49, in load
    return self.load_wsgiapp()
  File "/usr/local/lib/python3.7/site-packages/gunicorn/app/wsgiapp.py", line 39, in load_wsgiapp
    return util.import_app(self.app_uri)
  File "/usr/local/lib/python3.7/site-packages/gunicorn/util.py", line 358, in import_app
    mod = importlib.import_module(module)
  File "/usr/local/lib/python3.7/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1006, in _gcd_import
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 967, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 728, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/app/main.py", line 13, in <module>
    from .data import data_source
ImportError: attempted relative import with no known parent package
[2020-04-09 02:15:16 +0000] [10] [INFO] Worker exiting (pid: 10)
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/gunicorn/arbiter.py", line 209, in run
    self.sleep()
  File "/usr/local/lib/python3.7/site-packages/gunicorn/arbiter.py", line 357, in sleep
    ready = select.select([self.PIPE[0]], [], [], 1.0)
  File "/usr/local/lib/python3.7/site-packages/gunicorn/arbiter.py", line 242, in handle_chld
    self.reap_workers()
  File "/usr/local/lib/python3.7/site-packages/gunicorn/arbiter.py", line 525, in reap_workers
    raise HaltServer(reason, self.WORKER_BOOT_ERROR)
gunicorn.errors.HaltServer: <HaltServer 'Worker failed to boot.' 3>

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/bin/gunicorn", line 8, in <module>
    sys.exit(run())
  File "/usr/local/lib/python3.7/site-packages/gunicorn/app/wsgiapp.py", line 58, in run
    WSGIApplication("%(prog)s [OPTIONS] [APP_MODULE]").run()
  File "/usr/local/lib/python3.7/site-packages/gunicorn/app/base.py", line 228, in run
    super().run()
  File "/usr/local/lib/python3.7/site-packages/gunicorn/app/base.py", line 72, in run
    Arbiter(self).run()
  File "/usr/local/lib/python3.7/site-packages/gunicorn/arbiter.py", line 229, in run
    self.halt(reason=inst.reason, exit_status=inst.exit_status)
  File "/usr/local/lib/python3.7/site-packages/gunicorn/arbiter.py", line 342, in halt
    self.stop()
  File "/usr/local/lib/python3.7/site-packages/gunicorn/arbiter.py", line 393, in stop
    time.sleep(0.1)
  File "/usr/local/lib/python3.7/site-packages/gunicorn/arbiter.py", line 242, in handle_chld
    self.reap_workers()
  File "/usr/local/lib/python3.7/site-packages/gunicorn/arbiter.py", line 525, in reap_workers
    raise HaltServer(reason, self.WORKER_BOOT_ERROR)
gunicorn.errors.HaltServer: <HaltServer 'Worker failed to boot.' 3>
[2020-04-09 02:15:16 +0000] [12] [ERROR] Exception in worker process
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/gunicorn/arbiter.py", line 583, in spawn_worker
    worker.init_process()
  File "/usr/local/lib/python3.7/site-packages/uvicorn/workers.py", line 57, in init_process
    super(UvicornWorker, self).init_process()
  File "/usr/local/lib/python3.7/site-packages/gunicorn/workers/base.py", line 119, in init_process
    self.load_wsgi()
  File "/usr/local/lib/python3.7/site-packages/gunicorn/workers/base.py", line 144, in load_wsgi
    self.wsgi = self.app.wsgi()
  File "/usr/local/lib/python3.7/site-packages/gunicorn/app/base.py", line 67, in wsgi
    self.callable = self.load()
  File "/usr/local/lib/python3.7/site-packages/gunicorn/app/wsgiapp.py", line 49, in load
    return self.load_wsgiapp()
  File "/usr/local/lib/python3.7/site-packages/gunicorn/app/wsgiapp.py", line 39, in load_wsgiapp
    return util.import_app(self.app_uri)
  File "/usr/local/lib/python3.7/site-packages/gunicorn/util.py", line 358, in import_app
    mod = importlib.import_module(module)
  File "/usr/local/lib/python3.7/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1006, in _gcd_import
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 967, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 728, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/app/main.py", line 13, in <module>
    from .data import data_source
ImportError: attempted relative import with no known parent package
[2020-04-09 02:15:16 +0000] [12] [INFO] Worker exiting (pid: 12)
[2020-04-09 02:15:16 +0000] [11] [ERROR] Exception in worker process
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/gunicorn/arbiter.py", line 583, in spawn_worker
    worker.init_process()
  File "/usr/local/lib/python3.7/site-packages/uvicorn/workers.py", line 57, in init_process
    super(UvicornWorker, self).init_process()
  File "/usr/local/lib/python3.7/site-packages/gunicorn/workers/base.py", line 119, in init_process
    self.load_wsgi()
  File "/usr/local/lib/python3.7/site-packages/gunicorn/workers/base.py", line 144, in load_wsgi
    self.wsgi = self.app.wsgi()
  File "/usr/local/lib/python3.7/site-packages/gunicorn/app/base.py", line 67, in wsgi
    self.callable = self.load()
  File "/usr/local/lib/python3.7/site-packages/gunicorn/app/wsgiapp.py", line 49, in load
    return self.load_wsgiapp()
  File "/usr/local/lib/python3.7/site-packages/gunicorn/app/wsgiapp.py", line 39, in load_wsgiapp
    return util.import_app(self.app_uri)
  File "/usr/local/lib/python3.7/site-packages/gunicorn/util.py", line 358, in import_app
    mod = importlib.import_module(module)
  File "/usr/local/lib/python3.7/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1006, in _gcd_import
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 967, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 728, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/app/main.py", line 13, in <module>
    from .data import data_source
ImportError: attempted relative import with no known parent package
[2020-04-09 02:15:16 +0000] [11] [INFO] Worker exiting (pid: 11)
```

--- a/Dockerfile
+++ b/Dockerfile
@@ -1,19 +1,14 @@
 FROM tiangolo/uvicorn-gunicorn-fastapi:python3.7
 
-# ENVS RECOMENDATIONS
-ENV PYTHONDONTWRITEBYTECODE 1
-ENV PYTHONUNBUFFERED 1
-
-# PREPARE FOLDER
-WORKDIR /api
+ENV VARIABLE_NAME APP
 
 # COPY DEPENDENCIES
 COPY requirements.txt ./
 
+# COPY PROJECT
+COPY ./app /app/app
+
 # INSTALL DEPENDENCIES
 RUN pip install --no-cache-dir -r  requirements.txt
 
-# COPY PROJECT
-COPY . /api/
-
-CMD ["uvicorn", "--host", "0.0.0.0", "app.main:APP"]
+EXPOSE 80

--- a/README.md
+++ b/README.md
@@ -413,8 +413,17 @@ And don't despair if don't get the python setup working on the first try. No one
 
 ## Running / Development
 
+For a live reloading on code changes.
+
 * `pipenv run dev`
-* Visit your app at [http://localhost:8000](http://localhost:8000).
+
+Without live reloading.
+
+* `pipenv run start`
+
+Visit your app at [http://localhost:8000](http://localhost:8000).
+
+Alternatively run our API with Docker.
 
 ### Running Tests
 > [pytest](https://docs.pytest.org/en/latest/)
@@ -446,7 +455,22 @@ invoke generate-reqs
 
 [Pipfile.lock](./Pipfile.lock) will be automatically updated during `pipenv install`.
 
-### Building
+### Docker
+
+Our Docker image is based on [tiangolo/uvicorn-gunicorn-fastapi/](https://hub.docker.com/r/tiangolo/uvicorn-gunicorn-fastapi/).
+
+```bash
+invoke docker --build
+```
+
+Run with `docker run` or `docker-compose`
+
+### Invoke
+
+Additional developer commands can be run by calling them with the [python `invoke` task runner](http://www.pyinvoke.org/).
+```bash
+invoke --list
+```
 
 ### Deploying
 

--- a/docker-compose.yml
+++ b/docker-compose.yml
@@ -1,4 +1,4 @@
-version: '3.7'
+version: "3.7"
 
 services:
   api:
@@ -8,6 +8,6 @@ services:
     volumes:
       - .:/api
     ports:
-      - "8000:8000"
+      - "80:80"
     stdin_open: true
     tty: true

--- a/tasks.py
+++ b/tasks.py
@@ -9,6 +9,8 @@ Available commands
   invoke sort
   invoke check
 """
+import random
+
 import invoke
 
 TARGETS_DESCRIPTION = "Paths/directories to format. [default: . ]"
@@ -71,3 +73,17 @@ def generate_reqs(ctx):
     """Generate requirements.txt"""
     reqs = ["pipenv lock -r > requirements.txt", "pipenv lock -r --dev > requirements-dev.txt"]
     [ctx.run(req) for req in reqs]
+
+
+@invoke.task
+def docker(
+    ctx, build=False, run=False, tag="covid-tracker-api:latest", port=80, name=f"covid-api-{random.randint(0,999)}"
+):  # pylint: disable=too-many-arguments
+    """Build and run docker container."""
+    if not any([build, run]):
+        raise invoke.Exit(message="Specify either --build or --run", code=1)
+    if build:
+        docker_cmds = ["build", "."]
+    else:
+        docker_cmds = ["run", "-p", str(port), "--name", name]
+    ctx.run(" ".join(["docker", *docker_cmds, "-t", tag]))

--- a/tasks.py
+++ b/tasks.py
@@ -76,14 +76,12 @@ def generate_reqs(ctx):
 
 
 @invoke.task
-def docker(
-    ctx, build=False, run=False, tag="covid-tracker-api:latest", port=80, name=f"covid-api-{random.randint(0,999)}"
-):  # pylint: disable=too-many-arguments
+def docker(ctx, build=False, run=False, tag="covid-tracker-api:latest", name=f"covid-api-{random.randint(0,999)}"):
     """Build and run docker container."""
     if not any([build, run]):
         raise invoke.Exit(message="Specify either --build or --run", code=1)
     if build:
         docker_cmds = ["build", "."]
     else:
-        docker_cmds = ["run", "-p", str(port), "--name", name]
+        docker_cmds = ["run", "--publish", "80", "--name", name]
     ctx.run(" ".join(["docker", *docker_cmds, "-t", tag]))

